website/api/controllers/articles/view-basic-article.js

```diff
@@ -0,0 +1,79 @@
+module.exports = {
+
+  friendlyName: 'View blog article',
+
+
+  description: 'Display "Blog article" page.',
+
+  urlWildcardSuffix: 'slug',
+
+  inputs: {
+    slug : {
+      description: 'The relative path to the blog page from within this route.',
+      example: 'guides/deploying-fleet-on-render',
+      type: 'string',
+      defaultsTo: ''
+    }
+  },
+
+
+  exits: {
+    success: { viewTemplatePath: 'pages/articles/basic-article' },
+    badConfig: { responseType: 'badConfig' },
+    notFound: { responseType: 'notFound' },
+    redirect: { responseType: 'redirect' },
+  },
+
+
+  fn: async function ({slug}) {
+
+    if (!_.isObject(sails.config.builtStaticContent) || !_.isArray(sails.config.builtStaticContent.markdownPages) || !sails.config.builtStaticContent.compiledPagePartialsAppPath) {
+      throw {badConfig: 'builtStaticContent.markdownPages'};
+    }
+    let thisPage = _.find(sails.config.builtStaticContent.markdownPages, {
+      url: _.trimRight('/' + slug)
+    });
+    let needsRedirectMaybe = (!thisPage);
+
+    if (needsRedirectMaybe) {
+      // Creating a lower case, repeating-slashless slug
+      let multipleSlashesRegex = /\/{2,}/g;
+      let modifiedslug = slug.toLowerCase().replace(multipleSlashesRegex, '/');
+      // Finding the appropriate page content using the modified slug.
+      let revisedPage = _.find(sails.config.builtStaticContent.markdownPages, {
+        url: _.trimRight('/' + _.trim(modifiedslug, '/'), '/')
+      });
+      if(revisedPage) {
+        // If we matched a page with the modified slug, then redirect to that.
+        throw {redirect: revisedPage.url};
+      } else {
+        // If no page was found, throw a 404 error.
+        throw 'notFound';
+      }
+    }
+    // Setting the pages meta title and description from the articles meta tags.
+    // Note: Every article page will have a 'articleTitle' and a 'authorsFullName' meta tag.
```

Comment:
I think this should be `authorUsername` (no "s", because we don't normally use the possessive in variable names-- so that "s" can always be understood to indicate pluralization, and because "s" is an easy thing to omit on accident)

Reply:
Makes sense, I've changed `authorsGitHubUserName` to  `authorGitHubUsername` and  `authorsFullName` to  `authorFullName`.